src/plugins/intel_cpu/tests/functional/subgraph_tests/src/matmul_decompress_convert.cpp

```diff
@@ -0,0 +1,171 @@
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<MatMulDecompressConvertParams> obj) {
+        std::vector<InputShape> inputShapes;
+        std::pair<bool, bool> transpose;
+
+        std::tie(inputShapes, transpose) = obj.param;
+
+        std::ostringstream result;
+        for (const auto& shape : inputShapes) {
+            result << CommonTestUtils::partialShape2str({shape.first}) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                auto itr = shape.second.begin();
+                do {
+                    result << CommonTestUtils::vec2str(*itr);
+                } while (++itr != shape.second.end() && result << "_");
+            }
+            result << ")_";
+        }
+        result << "transpose_a=" << transpose.first << "_";
+        result << "transpose_b=" << transpose.second << "_";
+
+        return result.str();
+    }
+
+protected:
+    template<typename T>
+    void transposeShape(T& shape) {
+        IE_ASSERT(shape.size() > 1);
+        std::swap(*(shape.end() - 1), *(shape.end() - 2));
+    }
+
+    void SetUp() override {
+        abs_threshold = 0.01;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        std::vector<InputShape> inputShapes;
+        std::pair<bool, bool> transpose;
+
+        std::tie(inputShapes, transpose) = this->GetParam();
+
+        init_input_shapes(inputShapes);
+
+        bool transpA = transpose.first;
+        bool transpB = transpose.second;
+
+        if (transpA) {
+            transposeShape(inputDynamicShapes[0]);
+            for (auto& shapes : targetStaticShapes) {
+                transposeShape(shapes[0]);
+            }
+        }
+        if (transpB) {
+            transposeShape(inputDynamicShapes[1]);
+            for (auto& shapes : targetStaticShapes) {
+                transposeShape(shapes[1]);
+            }
+        }
+
+        const auto& inShapeA = inputDynamicShapes[0];
+        const auto& inShapeB = inputDynamicShapes[1];
+
+        std::string cpuNodeType = "FullyConnected";
+
+        auto params = builder::makeDynamicParams(element::f32, {inShapeA});
+        auto paramOuts = helpers::convert2OutputVector(helpers::castOps2Nodes<opset1::Parameter>(params));
+
+        auto matrixB = ngraph::builder::makeConstant<float16>(element::f16, inShapeB.get_shape(), {}, true);
+        auto convert = std::make_shared<ngraph::opset1::Convert>(matrixB, element::f32);
+        mark_as_decompression(convert);
+        auto matMul = builder::makeMatMul(paramOuts[0], convert, transpA, transpB);
+
+        function = CPUTestsBase::makeNgraphFunction(element::f32, params, matMul, cpuNodeType);
+    }
+};
+
+TEST_P(MatMulDecompressConvertTest, CompareWithRefs) {
+    run();
+    CheckNumberOfNodesWithType(compiledModel, "FullyConnected", 1);
+    CheckNumberOfNodesWithType(compiledModel, "Convert", 0);
```

Comment:
It would be great to add yet another check for Reorder. To my understanding it's not expected ether.

Reply:
added